test/integration/tests/button/message.js

```diff
@@ -0,0 +1,408 @@
+          message: {},
+          test: {
+            onRender() {
+              const frame = getElement("#testContainer iframe");
+              // $FlowFixMe
+              const win = frame.contentWindow;
+              const tagline = win.document.body.querySelector(CLASS.TAGLINE);
+
+              if (tagline) {
+                done(new Error(`Expected tagline not to render`));
+              }
+
+              assert.ok(getElementRecursive(".paypal-button-message-bottom"));
+              done();
+            },
+          },
+        })
+        .render("#testContainer");
+    });
+  });
+
+  describe(`placement`, () => {
+    describe("horizontal layout", () => {
+      it("should place message on bottom by default", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "horizontal",
+            },
+            message: {},
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-bottom"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+      it("should place message on top when position is top", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "horizontal",
+            },
+            message: {
+              position: "top",
+            },
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-top"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+      it("should place message on bottom when position is bottom", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "horizontal",
+            },
+            message: {
+              position: "bottom",
+            },
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-bottom"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+      it("should place message on bottom when no position is specified and credit/debit is a funding source", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "horizontal",
+            },
+            message: {},
+            fundingEligibility: {
+              credit: {
+                eligible: false,
+              },
+              paypal: {
+                eligible: true,
+              },
+              card: {
+                eligible: true,
+              },
+            },
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-bottom"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
```

Comment:
Is this just testing the same scenario as L172 since the horizontal layout will only show 2 buttons?

Reply:
now that you mention it, yes; I'll nix this test then